app/controllers/users/edit_mobile_controller.rb

```diff
@@ -0,0 +1,39 @@
+module Users
+  class EditMobileController < ApplicationController
+    include PhoneConfirmation
+
+    before_action :confirm_two_factor_authenticated
```

Comment:
Can you explain why? I don't think it matters, but maybe I'm missing something. `prepend_before_action` allows the method to run before any other `before_actions` that are defined. In most controllers where we use `confirm_two_factor_authenticated`, that's the only `before_action` we use, so there is no order to worry about.